src/unitxt/splitters.py

```diff
@@ -151,10 +149,12 @@ class DiverseLabelsSampler(Sampler):
 
     choices: str = "choices"
     labels: str = "labels"
+    random_generator: Random = None
 
     def prepare(self):
         super().prepare()
         self.labels_cache = None
+        self.random_generator = new_random_generator(sub_seed="diverse_labels_sampler")
```

Comment:
Same issue here with having the generator in the class scope.

Reply:
fixed